circuits/types/scalar/src/compare.rs

```diff
@@ -15,29 +15,51 @@
 // along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
 
 use super::*;
+use snarkvm_utilities::{FromBytes, ToBytes};
 
 impl<E: Environment> Compare<Scalar<E>> for Scalar<E> {
     type Boolean = Boolean<E>;
 
     /// Returns `true` if `self` is less than `other`.
     fn is_less_than(&self, other: &Self) -> Self::Boolean {
-        let mut is_less_than = Boolean::constant(false);
-        let mut are_previous_bits_equal = Boolean::constant(true);
-
-        // Initialize an iterator over `self` and `other` from MSB to LSB.
-        let bits_be = self.bits_le.iter().rev().zip_eq(other.bits_le.iter().rev());
-
-        for (index, (self_bit, other_bit)) in bits_be.enumerate() {
-            // Determine if `self` is less than `other` up to the `index`-th bit.
-            is_less_than |= &are_previous_bits_equal & (!self_bit & other_bit);
-
-            // Skip the update to the LSB, as this boolean is subsequently discarded.
-            if index != self.bits_le.len() - 1 {
-                are_previous_bits_equal &= self_bit.is_equal(other_bit);
+        // TODO: Will ScalarField ever outsize BaseField?
+        let scalar_modulus = match E::ScalarField::modulus().to_bytes_le() {
+            Ok(modulus_bytes) => match E::BaseField::from_bytes_le(&modulus_bytes) {
+                Ok(modulus) => modulus,
+                Err(error) => E::halt(format!("Failed to load the scalar modulus as a constant: {error}")),
+            },
+            Err(error) => E::halt(format!("Failed to retrieve the scalar modulus as bytes: {error}")),
+        };
+        if scalar_modulus.to_repr() <= E::BaseField::modulus_minus_one_div_two() {
```

Comment:
For us, this condition always holds. (We can add a debug_assert to ensure that.)